### ISSUE TITLE
_sufficientTags is broken

### ISSUE THREAD

**Issue by AJCM-git** [space-wizards/space-station-14#1038]
Right now we have this for `AccessReader` tags:
```
        private bool IsAllowed(List<string> accessTags)
        {
            foreach (var sufficient in _sufficientTags)
            {
                if (accessTags.Contains(sufficient))
                {
                    return true;
                }
            }
            foreach (var necessary in _necessaryTags)
            {
                if (!accessTags.Contains(necessary))
                {
                    return false;
                }
            }
            return true; <------ Problem
        }
```
The problem here is that when ` accessTags` does not contain `sufficient` it still returns true, making the airlock/locker with it accesible to every access level.

**Comment by ComicIronic** [comment]
Just to clarify: this is only a problem with sufficient is non-empty, right? If sufficient is empty, then returning true there is correct.

**Comment by Daracke** [comment]
Is this something that I'd be able to reproduce ingame, or is this more of a behind-the-scenes behavior with, I think, id cards and the reader itself?

**Comment by AJCM-git** [comment]
just place multiple access tag to an airlock with `sufficient` in `accessreader` like this:
```
- type: entity
  parent: AirlockCommand
  id: AirlockEVALocked	
  suffix:  Locked	 
  components:	  
    sufficient: ["Engineering", "External", "Command"]	   
```

**Comment by Daracke** [comment]
I might be getting confused. I changed the following within the following airlock. These are the red EVA airlock doors.

- type: entity
  parent: AirlockExternal
  id: AirlockExternalLocked
  suffix: External, Locked
  components:
  - type: AccessReader
    sufficient: ["Engineering", "External", "Command"]	<--Added this bit here
    necessary: ["External"]
And it's doing normal behavior and denying me access with an assistants card, and it's not letting me in.

I then removed the `necessary:` component so only sufficient exists, and _then_ it lets you in. So this is what letting anyone in?

**Comment by ComicIronic** [comment]
Yes, that's the issue. If the list of sufficient accesses is non-empty, then *not* having one of those accesses should deny entry.

**Comment by Daracke** [comment]
I've added the following bit between the foreach for the sufficient and the necessary blocks and maybe I got it working? 
 
 if (_necessaryTags.Count == 0)   // If the list doesn't exist, or exists and only has [""] then deny access.
            {
                return false;
            }
I'm still testing a bit, but it seems to keep access levels as expected if you straight remove the `necessary:` component line, or even just do `necessary: [""]` and leave it blank that way.

Let me mess around a little bit more, but I _think_ this should get it fine?
